tests/tutors/functional/test_materials_list_rst.py

```diff
@@ -0,0 +1,111 @@
+from collections.abc import AsyncIterator, Sequence
+from datetime import timezone
+
+import pytest
+from faker import Faker
+from starlette.testclient import TestClient
+
+from app.common.dependencies.authorization_dep import ProxyAuthData
+from app.tutors.models.materials_db import Material, MaterialKind
+from tests.common.active_session import ActiveSession
+from tests.common.assert_contains_ext import assert_response
+from tests.common.utils import remove_none_values
+from tests.tutors.factories import MaterialInputFactory
+
+pytestmark = pytest.mark.anyio
+
+
+MATERIALS_LIST_SIZE = 20
+
+
+@pytest.fixture()
+async def materials(
+    active_session: ActiveSession,
+    faker: Faker,
+    proxy_auth_data: ProxyAuthData,
+) -> AsyncIterator[Sequence[Material]]:
+    materials: list[Material] = []
+    async with active_session():
+        for _ in range(MATERIALS_LIST_SIZE):
+            materials.append(
+                await Material.create(
+                    **MaterialInputFactory.build_python(),
+                    tutor_id=proxy_auth_data.user_id,
+                    last_opened_at=faker.date_time_this_month(
+                        before_now=False,
+                        after_now=True,
+                        tzinfo=timezone.utc,
+                    )
+                )
+            )
+
+    materials.sort(key=lambda material: material.last_opened_at, reverse=True)
+
+    yield materials
+
+    async with active_session():
+        for material in materials:
+            await material.delete()
+
+
+@pytest.mark.parametrize(
+    "kind",
+    [
+        *[
+            pytest.param(member, id=member)
+            for member in MaterialKind.__members__.values()
+        ],
+        pytest.param(None, id="any_kind"),
+    ],
+)
+@pytest.mark.parametrize(
+    "limit",
+    [
+        pytest.param(1, id="limit_1"),
+        pytest.param(MATERIALS_LIST_SIZE // 2, id="limit_half"),
+        pytest.param(MATERIALS_LIST_SIZE // 4, id="limit_quarter"),
+    ],
+)
+@pytest.mark.parametrize(
+    "offset",
+    [
+        pytest.param(None, id="from_start"),
+        pytest.param(1, id="from_second"),
+        pytest.param(MATERIALS_LIST_SIZE // 2, id="from_middle"),
+        pytest.param(MATERIALS_LIST_SIZE, id="from_last"),
+    ],
+)
+async def test_materials_listing(
+    faker: Faker,
+    authorized_client: TestClient,
+    materials: Sequence[Material],
+    kind: MaterialKind | None,
+    offset: int | None,
+    limit: int,
+) -> None:
+
+    assert_response(
```

Comment:
polish: пустые строчки в начале классов/функций не ставятся

```suggestion
) -> None:
    assert_response(
```